modules/programs/vivid.nix

```
@@ -0,0 +1,141 @@
+{ config, lib, pkgs, ... }:
+let
+  yaml = pkgs.formats.yaml { };
+  bashLine = theme:
+    ''export LS_COLORS="$(${lib.getExe pkgs.vivid} generate ${theme})"'';
+  fishLine = theme:
+    "set -gx LS_COLORS (${lib.getExe pkgs.vivid} generate ${theme})";
+  nushellLine = theme: "${lib.getExe pkgs.vivid} generate ${theme}";
+  zshLine = bashLine;
+in {
+  meta.maintainers = [ lib.maintainers.arunoruto ];
+
+  options.programs.vivid = {
+    enable = lib.mkEnableOption ''
+      vivid - A themeable LS_COLORS generator with a rich filetype datebase
+      <link xlink:href="https://github.com/sharkdp/vivid" />
+    '';
+
+    package = lib.mkPackageOption pkgs "vivid" { };
+
+    enableBashIntegration = lib.mkOption {
+      default = true;
+      type = lib.types.bool;
+      description = ''
+        Whether to enable Bash integration.
+        Adds `${bashLine "<theme>"}` to Bash.
+      '';
+    };
+
+    enableFishIntegration = lib.mkOption {
+      default = true;
+      type = lib.types.bool;
+      description = ''
+        Whether to enable Fish integration.
+        Adds `${fishLine "<theme>"}` to Fish.
+      '';
+    };
+
+    enableNushellIntegration = lib.mkOption {
+      default = true;
+      type = lib.types.bool;
+      description = ''
+        Whether to enable Nushell integration.
+        Adds `${nushellLine "<theme>"}` to Nushell.
+      '';
+    };
+
+    enableZshIntegration = lib.mkOption {
+      default = true;
+      type = lib.types.bool;
+      description = ''
+        Whether to enable Zsh integration.
+        Adds `${zshLine "<theme>"}` to Zsh.
+      '';
+    };
+
+    theme = lib.mkOption {
+      type = lib.types.nullOr lib.types.str;
```

```suggestion
      type = with types; nullOr str;
```

assuming that `with lib;` was declared above.

---

The same goes for the other options as well.

---

I have rewritten the module with `with lib;`.
I checked it multiple times, so there shouldn't be anything missing.